### PR TITLE
[DataGrid] warn devs when `autoPageSize` used with `autoheight`

### LINKED ISSUE #3233: [DataGrid] No rendering with autoHeight + autoPageSize
### Current behavior 😯

The grid does not render. Weird `1-0 of XXX` in the footer

### Expected behavior 🤔

The grid renders, filling availble height (because of `autoHeight`) and showing all rows

### Steps to reproduce 🕹

Code sandbox using v5.0.1, exposing the issue: https://codesandbox.io/s/datagrid-v5-0-1-autoheight-autopagesize-9dy64?file=/src/App.tsx

Same one using v5.0.0, rendering normally: https://codesandbox.io/s/datagrid-v5-0-0-autoheight-autopagesize-working-hks4q?file=/src/App.tsx

### Context 🔦

I understand that autoHeight + autoPageSize probably can't work together (or can they?) but if that's the case, can you please just ignore autoPageSize, or log a warning or whatever? Simply failing to render is not a good user experience.

As suggested by @flaviendelangle it might be a regression of #3007 

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: Linux 5.13 Linux Mint 20.2 (Uma)
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: Not Found
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 96.0.4664.45
    Firefox: 94.0

Used browser: Chrome
```
</details>

### Order ID 💳 (optional)

I have a pro plan, but don't have access to the order id right now :)

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Closes #3233

### CODE REVIEW

**Comment by romgrk** [comment]
I don't think doing it this way will be possible, we autogenerate the proptypes from the typescript definitions. I think if we add the proptype check, you'd need to do it after the `Component.propTypes = ...` section is defined as to not interfere with the autogeneration.

**Comment by Sboonny** [comment]
Thank you for the quick context, unfortunately I still don't get it, where exactly is after it? 

Just for context, I have looked up `checkboxSelectionVisibleOnly` to mimic its logic in [DataGridProps.ts file](https://github.com/mui/mui-x/blob/19b9eb541f64885bccd47b65dd80360d6eff17f6/packages/grid/x-data-grid/src/models/props/DataGridProps.ts#L139C40-L139C40), I found normal type declaration, so I thought all I need is to add `chainPropTypes` and run `yarn proptypes` to make sure it doesn't change, but I am completely wrong, and any additional help is appreciated.

**Comment by flaviendelangle** [comment]
The proptypes containing `chainPropTypes` are preserved during the generation, the current approach should work fine on this regard :+1:

**Comment by romgrk** [comment]
```js
Component.propTypes = /* autogenerated prop types */

// manually written prop type
Component.propTypes.autoPageSize = chainProps(Component.propTypes.autoPageSize, ...)
```

But I'm not sure how we feel about adding prop types like that, I'm fine with it but maybe somone in @mui/xgrid has a different opinion?

**Review by romgrk — APPROVED** [review]

File: packages/grid/x-data-grid/src/tests/rows.DataGrid.test.tsx [hunk 1]
```diff
@@ -1111,6 +1111,14 @@ describe('<DataGrid /> - Rows', () => {
       act(() => apiRef.current.updateRows([{ id: 5, brand: 'Atari' }]));
       expect(getColumnValues(0)).to.deep.equal(['Apple', 'Atari']);
     });
+
+    it('should throw a console error if autoPageSize is used with autoHeight', () => {
+      expect(() => {
+        render(<TestCase autoPageSize autoHeight />);
+      }).toErrorDev(
+        'MUI: The `autoPageSize` prop will conflict with `autoHeight` prop when both of them are enabled.',
+      );
```
Comment (oliviertassinari) [inline packages/grid/x-data-grid/src/tests/rows.DataGrid.test.tsx:1120]: On the warning message, I would recommend to follow https://www.notion.so/mui-org/Technical-writing-guidance-7e55b517ac2e489a9ddb6d0f6dd765de?pvs=4#85219822c3194b6f8a49ee08ea82b90a.

- Explain what happened, e.g. you provided autoPageSize={true} and autoHeight={true}
- Explain why it's wrong, e.g. these two props can't function together.
- Provide a way out, e.g. please remove one of these two props.

Comment (Sboonny) [reply packages/grid/x-data-grid/src/tests/rows.DataGrid.test.tsx:1120]: is it fine, if I update the error message with the type change PR?

**Pull request merged**